client/interface.go

```diff
@@ -0,0 +1,33 @@
+package client
+
+import (
+	"encoding"
+
+	"github.com/libp2p/go-libp2p-core/event"
+)
+
+// Client represents the drand Client interface.
+type Client interface {
+	// TextMarshaler (the 'MarshalText' method) allows serialization of the client
+	// state for subsequent restoration. The marshalled value can be passed to `New`
+	// to restore Client state.
+	encoding.TextMarshaler
+
+	// Get returns a the randomness at `round` or an error over a result channel.
+	// One result will be returned, with exactly one of Err or Data populated.
+	func Get(ctx context.Context, round uint64) <-chan Result
```

Comment:
I would prefer to avoid forcing a channel/async on the caller. A caller can always introduce channel semantics if they want, but dealing with it is a pain, and error-prone, if unwanted. I expect that the _usual_ case here is that the value will be already cached after being received on a pubsub, and returned from memory. The caller may well want a clear signal if that's not the case.

As a concrete alternative:
```
// Returns an entry, blocks until it's available.
func Get(ctx, round) (BeaconEntry, error)
```
where the caller can effect a time-out through the context, including getting synchronous semantics with an already-cancelled context. 

Alternatively, an explicit timeout argument can be added, and the ctx-wrapping done internally. This has both advantages of convenience and disadvantages in flexibility.
```
// Returns an entry, blocks until it's available or the deadline expires
func Get(ctx, round, deadline time.Time) (BeaconEntry, error)
```

Another alternative, make the synchronous and async paths explicit:
```
// Returns an entry, or an error if unavailable
func Get(ctx, round) (BeaconEntry, error)
// Waits for an entry to be available (or ctx to be cancelled)
func Await(ctx, round) (BeaconEntry, error) // or Fetch(), Request(), ...
```